File: string_api.py
s = 'hello' + ' world'
print "3.---length of '"+s+"' is "+str(len(s))
print "4.---first 5 characters of '"+s + "' is '" + s[0:5] + "'"
print "4.1---last 5 characters of '"+s + "' is '" + s[-5:] + "'"
print "5.---position of 'world' in '"+s + "' is " + str(s.index("world"))
print "6.---replace 'world' by 'moon' in '"+s + "' is " + s.replace("world","moon",1)
print "6.1---replace all 'l' by '?' in '"+s + "' is " + s.replace("l","?")
print "7.---split '"+s + "' by space is ",s.split(" ")
print "8.---reverse of '"+s + "' is ",s[::-1]
print "8.1---reverse of '"+s + "' is ",''.join(reversed(s))
print "9.---uppercase of '"+s + "' is ",s.upper()
s1 = "Hello World"
print "10.---lowercase of '"+s1 + "' is ",s1.lower()
print "11.---capitalize of '"+s1 + "' is ",s1.capitalize()
s2 = "hello world"
print "11.---compare '"+s + "' with '" + s2 + "' returns ", s==s2
print "11.1---compare '"+s + "' with '" + s1 + "' returns ", s==s1
print "12.--- iterate over each char of '%s'" % s
for ch in s:
  print ch

File: string_api.rb
puts "1.--- string literal"
s='hello world'; puts "string constant:",s
t="\tsay #{s}\n\tthen say bye"; puts "string template:",t
multilines=<<EOS
Hello
World
EOS
puts "mulilines:",multilines
puts "2.--- append 'hello' with ' world' =#{'hello' + ' world'}"
puts "3.--- length of #{s}=#{s.size}"
puts "4.--- first 5 character of #{s}=#{s[0..4]}"
puts "4.1--- last 5 character of #{s}=#{s[-5..-1]}"
puts "5.--- position of world in #{s}=#{s.index("world")}"
puts "5.1--- position of moon in #{s}=#{s.index("moon").inspect}"
puts "6.--- replace world by moon in #{s}=#{s.sub("world","moon")}"
puts "6.1--- replace all 'l' by '?' in #{s}=#{s.gsub("l","?")}"
puts "7.--- split  #{s} by space =#{s.split(" ").inspect}"
puts "8.--- reverse of  #{s}=#{s.reverse}"
puts "9.--- uppercase of  #{s}=#{s.upcase}"
puts "10.--- lowercase of  #{s}=#{s.downcase}"
s1 = 'hello world'
puts "11.--- compare #{s} with #{s1} returns #{s == s1}"
s2 = 'hello World'
puts "11.1--- compare #{s} with #{s2} returns #{s == s2}"
puts "12. --- iterate over each char"
require 'jcode'; s2.each_char{ |ch| puts ch }
puts "12.1 --- iterate over each char"
s2.each_byte {|b| puts b.chr}
puts "12.2 --- iterate over each char"
s2.split('').each {|ch| puts ch}
